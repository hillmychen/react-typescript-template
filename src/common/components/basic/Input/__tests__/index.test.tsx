import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import { Input } from '../index'

test('Input onChange Event', () => {
  const { asFragment, getByRole } = render(<Input />)
  const input = getByRole('textbox', { hidden: true }) as HTMLInputElement
  fireEvent.change(input, { target: { value: 'test value' } })
  expect(input.value).toEqual('test value')
  expect(input)
  expect(asFragment()).toMatchSnapshot()
})

test('Input test trim', done => {
  const { asFragment, getByRole } = render(<Input trim onChange={handlerOnChange} />)
  const input = getByRole('textbox', { hidden: true }) as HTMLInputElement
  fireEvent.change(input, { target: { value: ' test value ' } })
  function handlerOnChange(event: React.ChangeEvent<HTMLInputElement>) {
    expect(event.target.value).toEqual('test value')
    expect(asFragment()).toMatchSnapshot()
    done()
  }
})
