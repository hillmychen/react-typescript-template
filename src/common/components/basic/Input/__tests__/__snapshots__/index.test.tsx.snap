// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input onChange Event 1`] = `
<DocumentFragment>
  <input
    class="ant-input"
    type="text"
    value="test value"
  />
</DocumentFragment>
`;

exports[`Input test trim 1`] = `
<DocumentFragment>
  <input
    class="ant-input"
    type="text"
    value=""
  />
</DocumentFragment>
`;
